src/module/macros/attack.ts

```diff
@@ -0,0 +1,593 @@
+  });
+  // Warn about missing damage type if the value is non-zero
+  if (typeMissing) {
+    ui.notifications!.warn(`Warning: ${item.name} has a damage value without type!`);
+  }
+
+  // Options processing
+  if (options) {
+    if (options.accBonus) {
+      mData.grit += options.accBonus;
+    }
+    if (options.damBonus) {
+      let i = mData.damage.findIndex(dam => {
+        return dam.DamageType === options.damBonus.type;
+      });
+      if (i >= 0) {
+        // We need to clone so it doesn't go all the way back up to the weapon
+        let damClone = { ...mData.damage[i] };
+        if (parseInt(damClone.Value) > 0) {
+          damClone.Value = `${damClone.Value}+${options.damBonus.val}`;
+        } else {
+          damClone.Value = options.damBonus.val.toString();
+        }
+        // @ts-expect-error Not the full class, but it should work for our purposes.
+        mData.damage[i] = damClone;
+      } else {
+        // @ts-expect-error Not the full class, but it should work for our purposes.
+        mData.damage.push({ Value: options.damBonus.val.toString(), DamageType: options.damBonus.type });
+      }
+    }
+  }
+  // Check if weapon if loaded.
+  if (getAutomationOptions().limited_loading && getAutomationOptions().attacks) {
+    if (is_loading(itemEnt) && !itemEnt.Loaded) {
+      ui.notifications!.warn(`Weapon ${item.data.data.name} is not loaded!`);
+      return;
+    }
+    if (is_limited(itemEnt) && itemEnt.Uses <= 0) {
+      ui.notifications!.warn(`Weapon ${item.data.data.name} has no remaining uses!`);
+      return;
+    }
+    if (mData.destroyed) {
+      ui.notifications!.warn(`Weapon ${item.data.data.name} is destroyed!`);
+      return;
+    }
+  }
+
+  // Prompt the user before deducting charges.
+  const targets = Array.from(game!.user!.targets);
+  let { AccDiffData } = await import("../helpers/acc_diff");
+  const initialData =
+    rerollData ??
+    AccDiffData.fromParams(item, mData.tags, mData.title, targets, mData.acc > 0 ? [mData.acc, 0] : [0, -mData.acc]);
+
+  let promptedData;
+  try {
+    let { open } = await import("../helpers/slidinghud");
+    promptedData = await open("attack", initialData);
+  } catch (_e) {
+    return;
+  }
+
+  const atkRolls = attackRolls(mData.grit, promptedData);
+
+  // Deduct charge if LOADING weapon.
+  if (getAutomationOptions().limited_loading && getAutomationOptions().attacks) {
+    if (is_loading(itemEnt)) {
+      itemEnt.Loaded = false;
+      await itemEnt.writeback();
+    } else if (is_limited(itemEnt)) {
+      itemEnt.Uses = itemEnt.Uses - 1;
+      await itemEnt.writeback();
+    }
+  }
+
+  let rerollMacro = {
+    title: "Reroll attack",
+    fn: "prepareEncodedAttackMacro",
+    args: [actor.data.data.derived.mm!.as_ref(), item.id, options, promptedData.toObject()],
+  };
+
+  await rollAttackMacro(actor, atkRolls, mData, rerollMacro);
+}
+
+export async function openBasicAttack(rerollData?: AccDiffData) {
```

Comment:
A possibility for downsizing this file further would be separating the handlers for the attack HUD. @SohumB, thoughts?

At a glance, that would include `checkTargets`, `openBasicAttack`, `applyPluginsToRoll`, and maybe `attackRolls`.

Reply:
Fine by me, though it is difficult to figure out a clean separation here between "attack-related things" and "things the attack HUD uses". I'd suggest getting it down to a 600-line file is a great first step, and we can reorg that file the next time someone needs to actually touch it.